### PR TITLE
add unit tests for cache statistics controller

### PR DESCRIPTION
**Pull request by sheridancbio**
- mock CacheStatisticsService to throw CacheNotFoundException when requests for unknown cache alias are made
- store requested cache name in CacheNotFoundException
- test keysInCache endpoint and keyCountsPerClass endpoints with valid and invalid cache aliases
adjust GlobalExceptionHandler to return a 404 NotFound status for requests on an invalid cache alias

### CODE REVIEW

File: web/src/main/java/org/cbioportal/web/error/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -177,4 +177,12 @@
         ErrorResponse response = new ErrorResponse("Specified token cannot be found");
         return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(CacheNotFoundException.class)
+    public ResponseEntity<ErrorResponse> handleCacheNotFoundException(CacheNotFoundException ex) {
+
```
Comment (ao508) [inline web/src/main/java/org/cbioportal/web/error/GlobalExceptionHandler.java]: extra line here :)

File: web/src/test/java/org/cbioportal/web/CacheStatsControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
```
Comment (ao508) [inline web/src/test/java/org/cbioportal/web/CacheStatsControllerTest.java]: extra new line

File: web/src/main/java/org/cbioportal/web/error/GlobalExceptionHandler.java (hunk 1 above)
Comment (sheridancbio) [reply web/src/main/java/org/cbioportal/web/error/GlobalExceptionHandler.java]: Ok .. I will clean it up here, and also in the other handler declarations above.

Comment (sheridancbio) [reply web/src/main/java/org/cbioportal/web/error/GlobalExceptionHandler.java]: done

File: web/src/test/java/org/cbioportal/web/CacheStatsControllerTest.java (hunk 2 above)
Comment (sheridancbio) [reply web/src/test/java/org/cbioportal/web/CacheStatsControllerTest.java]: fixed .. thanks

**Pull request merged**